psi4/src/psi4/libscf_solver/hf.cc

```diff
@@ -720,23 +724,34 @@ void HF::find_occupation()
     } else {
         std::vector<std::pair<double, int> > pairs_a;
         std::vector<std::pair<double, int> > pairs_b;
-        for (int h=0; h<epsilon_a_->nirrep(); ++h) {
-            for (int i=0; i<epsilon_a_->dimpi()[h]; ++i)
+        for (int h = 0; h < epsilon_a_->nirrep(); ++h) {
+            for (int i = 0; i < epsilon_a_->dimpi()[h]; ++i){
                 pairs_a.push_back(std::make_pair(epsilon_a_->get(h, i), h));
+            }
         }
-        for (int h=0; h<epsilon_b_->nirrep(); ++h) {
-            for (int i=0; i<epsilon_b_->dimpi()[h]; ++i)
-                pairs_b.push_back(std::make_pair(epsilon_b_->get(h, i), h));
+        sort(pairs_a.begin(), pairs_a.end());
+
+        // Do we need to sort beta?
```

Comment:
@fevangelista @CDSherrill @andysim I think this should fix #738. We simply take the sort of `pairs_a` if we are a singlet, as I believe we effectively have a RHF reference for UHF/CUHF/ROHF reference singlets. This will ensure the resulting code is always a singlet regardless of degenerate noise in the sort. I think this handles all cases, but some thought to make sure this doesn't trip up an edge case is required.